povm_toolbox/library/randomized_projective_measurements.py

```diff
@@ -193,35 +248,33 @@ def to_sampler_pub(
         circuit: QuantumCircuit,
         circuit_binding: BindingsArray,
         shots: int,
+        *,
         pass_manager: StagedPassManager | None = None,
     ) -> tuple[SamplerPub, RPMMetadata]:
         """Append the measurement circuit(s) to the supplied circuit.
 
-        This method takes a supplied circuit and append the measurement circuit
-        to it. As the measurement circuit is parametrized, its parameters values
-        are concatenated with the parameter values associated with the supplied
-        quantum circuit. TODO: explain how the distribution of the shots is done.
-        If a randomized POVM is used, the end user's parameters have to be
-        concatenated with the sampled POVM parameters. The POVM parameters are a
-        2-D (TODO: update docstring in next PR, which will change this method anyways)
+        This method takes a supplied circuit and appends the measurement circuit(s) to it. If the
+        measurement circuit is parametrized, its parameters values should be concatenated with the
+        parameter values associated with the supplied quantum circuit.
+
+        .. warning::
+           The actual number of measurements executed will depend not only on the provided ``shots``
+           value but also on the value of :attr:`.shot_repetitions`.
 
         Args:
             circuit: A quantum circuit.
             circuit_binding: A bindings array.
-            shots: A specific number of shots to run with. Note that ``shots`` is
-                effectively the number of times we sample PVMs. Then, for each
-                sampled PVM, the measurement is repeated ``self.shot_repetitions``
-                times. Therefore, the actual total number of measurement shots is
-                then ``shots`` multiplied by ``self.shot_repetitions``.
-            pass_manager: An optional pass manager. After the supplied circuit has
-                been composed with the measurement circuit, the pass manager will
+            shots: A specific number of shots to run with.
+            .. note::
+               test
```

Comment:
"note" to be completed ?

Reply:
Oh oops :smile: I was testing whether I could include a `.. note` admonition here to document the relation of `shots` and `shot_repetitions` but I ended up moving that higher up in the docstring